receipt/src/main/java/com/hyperwallet/android/receipt/view/ReceiptDetailFragment.java

```diff
@@ -0,0 +1,136 @@
+import com.hyperwallet.android.model.receipt.Receipt;
+import com.hyperwallet.android.receipt.R;
+import com.hyperwallet.android.receipt.viewmodel.ReceiptDetailViewModel;
+
+import java.text.DecimalFormat;
+
+public class ReceiptDetailFragment extends Fragment {
+
+    private ReceiptDetailViewModel mReceiptDetailViewModel;
+
+    public ReceiptDetailFragment() {
+        setRetainInstance(true);
+    }
+
+    public static ReceiptDetailFragment newInstance() {
+        ReceiptDetailFragment fragment = new ReceiptDetailFragment();
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mReceiptDetailViewModel = ViewModelProviders.of(requireActivity()).get(ReceiptDetailViewModel.class);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+            Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_receipt_detail, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        Receipt receipt = mReceiptDetailViewModel.getReceipt();
+
+        // transactions
+        ReceiptViewUtil util = new ReceiptViewUtil();
+        util.setTransactionView(receipt, view);
+
+        // fee details
+        if (!TextUtils.isEmpty(receipt.getFee())) {
+            view.findViewById(R.id.fee_details_layout).setVisibility(View.VISIBLE);
+            double feeAmount = Double.parseDouble(receipt.getFee());
+            double transferAmount = Double.parseDouble(receipt.getAmount());
+            double amount = transferAmount - feeAmount;
+            DecimalFormat decimalFormat = new DecimalFormat(AMOUNT_FORMAT);
+
+            TextView amountView = view.findViewById(R.id.details_amount_value);
+            amountView.setText(view.getContext().getString(R.string.amount_view_format,
+                    decimalFormat.format(amount), receipt.getCurrency()));
+
+            TextView fee = view.findViewById(R.id.details_fee_value);
+            fee.setText(view.getContext().getString(R.string.amount_view_format,
+                    decimalFormat.format(feeAmount), receipt.getCurrency()));
+
+            TextView transfer = view.findViewById(R.id.details_transfer_amount_value);
+            transfer.setText(view.getContext().getString(R.string.amount_view_format,
+                    decimalFormat.format(transferAmount), receipt.getCurrency()));
+        }
+
+        // receipt details
+        TextView receiptId = view.findViewById(R.id.receipt_id_value);
+        receiptId.setText(receipt.getJournalId());
+        TextView date = view.findViewById(R.id.date_value);
+        date.setText(DateUtils.toDateFormat(DateUtils.
+                fromDateTimeString(receipt.getCreatedOn()), DETAIL_DATE_FORMAT));
+
```

Comment:
@peterm What about extracting ReceiptDetails to simplify the code:
ReceiptDetails details = receipt.getDetails();
if (details == null){ return; }
 if (!TextUtils.isEmpty(details.getCharityName())) {
....